### ISSUE TITLE
unsafeClient not ready yet error

### ISSUE THREAD

**Issue by jasonblewis** [matrix-org/matrix-appservice-irc#944]
at LCA2020 we have a few people seeing this message when they try and connect to the bridge to freenode.

The bridge reports unsafeClient not ready yet 

this issue results in riot being able to join the irc channel and see what is happening on the channel but unable to send to the channel. when you send a message it appears to send correctly, but it never arrives in the irc channel, with no visible error message.

see attached screen shot

**Comment by jasonblewis** [comment]
![unsafeClient not yet ready1](https://user-images.githubusercontent.com/129165/72317656-12939680-36ee-11ea-9635-484a4b2e4ba7.png)
![unsafeClient not yet ready2](https://user-images.githubusercontent.com/129165/72317667-1a533b00-36ee-11ea-8cd4-f331bb51d1d9.png)

**Comment by madduck** [comment]
I should note that this happens both with the explicit integration in `#linux.conf.au:matrix.org`, as well as the implicitly bridged froom `#freenode_#linux.conf.au:matrix.org`, and it's a problem for a couple of people, while it works for plenty others at the same time.

A little bit more context: we used to have issues with `err_reggednick` until I removed `+r` from the IRC channel yesterday. Before that, `@appservice-irc:matrix.org` would report `err_reggednick` to some users, but not to others. It's possible that there's a correlation here too, i.e. only those for whom the bridge held a connection to IRC would get the error, whereas those who weren't connected, consequently wouldn't see the error.

Is there a way to force (re-)connection? The screenshot shows various different responses, but there's no `!connect` or `!reconnect`, according to the source code.

I should also point out that in the case of @jasonblewis (k-man), we tried `!quit`, subsequently got kicked out of the implicitly-bridged room, left the chat with `@appservice-irc:matrix.org`, and started from scratch (having to manually start a new chat with `appservice-irc:matrix.org`), but nothing seems to have changed.

**Comment by mattcen** [comment]
I'm experiencing the same issue.

<img width="304" alt="image" src="https://user-images.githubusercontent.com/4273659/72318330-b62d7880-36e7-11ea-8317-181413f1c073.png">

**Comment by nazunalika** [comment]
I am having the same issues here as well. 

![image](https://user-images.githubusercontent.com/11469736/72670510-42e37980-39fb-11ea-9c6a-efc70b87260c.png)
![image](https://user-images.githubusercontent.com/11469736/72670526-68708300-39fb-11ea-8ef3-59d25a63ee1e.png)

```
14:05 -!- There is no such nick Sokel[m]
14:05 -!- There is no such nick nazunalika
```

**Comment by jasonblewis** [comment]
See also #926

**Comment by tilosp** [comment]
The same is happening to me

**Comment by Gnurou** [comment]
Same problem, happened as soon as I tried using the bridge. There doesn't seem to be any workaround besides restarting the service, which is not an option unless you are running your own instance.

**Comment by vs49688** [comment]
Same issue happened to me last night.
In my case I changed my `!storepass` using a grouped nick instead of a primary one. After this, `NickServ` stopped responding. Further investigation yielded the same responses seen here.

Various combinations of `!removepass` and `!quit` yield no change.

Strangely, `!nick` is still returning the old grouped (non-primary) username. I can verify it's not in use via logging in with a native IRC client and talking to `NickServ` that way. 

When talking to the appservice user:

```
$ !nick
Format: '!nick DesiredNick' or '!nick irc.server.name DesiredNick'
Currently connected to IRC networks:
chat.freenode.net as mynick

$ !nick primarynick
You are not connected to the network.

$ !whois mynick
unsafeClient not ready yet

$ !listrooms
You are connected, but not joined to any channels.

$ !bridgeversion
BridgeVersion: 0.16.0
```

It also appears this bloke's been hit by the same issue: https://github.com/matrix-org/matrix-appservice-irc/issues/1006

**Comment by ocdtrekkie** [comment]
Definitely unable to get the IRC bridge working with Freenode as a new user. Seeing the exact behaviors described here.

**Comment by PureBasic** [comment]
I tried, and tried, and tried, and tried...
![image](https://user-images.githubusercontent.com/8854321/79058989-eea40a80-7c74-11ea-9f38-683d8fc68649.png)
At least I'm not alone :)

**Comment by Gnurou** [comment]
For me it started working again a few days ago (I guess because the bridge has been restarted or something?). I have been able to use the service normally since then.

The issue mentioned above ("Reconnect after !storepass, even if the user isn't in any channels") seems to imply that the service will reconnect if you are in a channel, maybe try to join any channel and do a `storepass` again?

**Comment by PureBasic** [comment]
I cannot join any IRC channel. I get kicked with `Reason: IRC connection failure.`

**Comment by PureBasic** [comment]
Found where my problem might have been coming from. I had two '!' exclamation points in my password. It was also causing problems with weechat. I changed my pass to something simpler and was able to connect on Riot also.
This password was working with kiwi and pidgin.

**Comment by benbullard79** [comment]
I'm experiencing the issue exactly as described in first two posts.

**Comment by benbullard79** [comment]
!cmd chat.freenode.net PING
Error: Possibly disconnected

So far !quit command will not force bridge to quit. It returns nothing exactly as is it were "Possibly disconnected"

And the other commands in second post bring the same results.

**Comment by benbullard79** [comment]
My problem seems to have healed itself.

Faith healing?

**Comment by ailin-nemui** [comment]
this is still happening. the missing failure indication of sending messages is most confusing. sometimes it self-heals after many hours...

**Comment by jchv** [comment]
Looks like it's still happening. Unfortunately, it hit me as a very new Matrix user, the first time I tried to use the IRC bridge. Is it possible I did something wrong?

**Comment by nekr0z** [comment]
Still an issue.

![image](https://user-images.githubusercontent.com/1515080/84023106-2b982f00-a990-11ea-824c-b9011b1bc9ef.png)

**Comment by tilosp** [comment]
@nekr0z does `!quit` output anything?

**Comment by jchv** [comment]
I can say that `!quit` does not work or output anything in my case.

**Comment by nekr0z** [comment]
> @nekr0z does `!quit` output anything?

Nope, no reaction to `!quit` at all.

**Comment by tilosp** [comment]
Then this will remain broken for you until the next restart of the bridge, normally `!quit` should output `You are not connected to any networks.` the second time you run it.

I have a [PR](https://github.com/matrix-org/matrix-appservice-irc/pull/976) open, that should at least allow you to disconnect using `!guit` and reconnect afterwards.

**Comment by nekr0z** [comment]
So, this morning I tried to check if the bridge got restarted and I could use IRC in Matrix.

![image](https://user-images.githubusercontent.com/1515080/84108213-55049980-aa28-11ea-91dc-e70672bb324d.png)

Oops, that makes sense, let me log out of the IRC client first and try again...

![image](https://user-images.githubusercontent.com/1515080/84108306-8c734600-aa28-11ea-943b-ef892e09aa08.png)

Hmm... Really?

![image](https://user-images.githubusercontent.com/1515080/84108394-c0e70200-aa28-11ea-90e3-d563613e7c18.png)

Again? Yes. Again. Since this, all the same symptoms as yesterday.

Is it some kind of prank?

**Comment by Nuc1eoN** [comment]
I just installed Riot and I am hitting the same issue.

The real PITA however is, that messages appear to be send. I was lucky to still have hexchat open aside, which allowed to to notice that my message are actually not being sent to the IRC channel.

**Comment by slondr** [comment]
Can confirm that I also am having this issue now

**Comment by Half-Shot** [comment]
This should hopefully have been fixed in #1098, is anyone still experiencing this on matrix.org?

**Comment by penn5** [comment]
> 
> 
> This should hopefully have been fixed in #1098, is anyone still experiencing this on matrix.org?

I am experiencing this. !storepass appears to work, !quit has no response, !cmd chat.freenode.net PING gives Error: Client is not connected and !nick gives, of course, unsafeClient not ready yet. I signed up for my account today. People can see my messages on IRC and I can see theirs.

**Comment by Half-Shot** [comment]
The voices in my head say no.
